pkgs/development/libraries/speexdsp/default.nix

```diff
@@ -1,25 +1,21 @@
-{ lib, stdenv, fetchurl, autoreconfHook, pkg-config, fftw }:
+{ lib, stdenv, fetchurl, autoreconfHook, pkg-config }:
 
 stdenv.mkDerivation rec {
   pname = "speexdsp";
   version = "1.2.1";
 
   src = fetchurl {
     url = "https://downloads.xiph.org/releases/speex/${pname}-${version}.tar.gz";
-    sha256 = "sha256-jHdzQ+SmOZVpxyq8OKlbJNtWiCyD29tsZCSl9K61TT0=";
+    hash = "sha256-jHdzQ+SmOZVpxyq8OKlbJNtWiCyD29tsZCSl9K61TT0=";
   };
 
-  patches = [ ./build-fix.patch ];
   postPatch = "sed '3i#include <stdint.h>' -i ./include/speex/speexdsp_config_types.h.in";
 
   outputs = [ "out" "dev" "doc" ];
 
   nativeBuildInputs = [ autoreconfHook pkg-config ];
-  buildInputs = [ fftw ];
 
-  configureFlags = [
-    "--with-fft=gpl-fftw3"
```

Comment:
Maybe we should add a flag to this package instead, like `withFftw3`, and override this flag only for EasyEffects.

Reply:
I am not sure about this, the package was added to nixpkgs with fftw, there is no record on github why this was done. No other distro that I have seen builds with fftw, not using fftw doesn't cause any missing functionality, it just uses another fft library. The fact that easyeffects explodes in this case makes me think that speexdsp might be slightly broken when using fftw. Perhaps other application that use it are somewhat broken too.
Another strange thing is that when fftw support was added it was marked as experimental.
https://gitlab.xiph.org/xiph/speexdsp/-/commit/3ea24daa971f5aa64f522c4d5a30988e35cf55bc
And 2 days later it wasn't experimental anymore
https://gitlab.xiph.org/xiph/speexdsp/-/commit/2fee1222850076810d0ea575a66a9596a4b807a7

Reply:
One concern that I have is that `speex` also uses FFTW here in nixpkgs (it can be seen in the `all-packages.nix`) and I am not sure if there is a reason for it or not. So with the flag it would be easier to limit the impact of the change right now, and if there is a good reason to disable it in all packages it will be easy too.

Reply:
done